### ISSUE TITLE
SERIALOPEN, SERIALPRINT issue

### ISSUE THREAD

**Issue by shoogend** [CUASAS/pixel-gantry-control#83]
I am able to connect with SERIALOPEN (i.e. there is no error message in the gScript application), but occasionally when I attempt to send commands through SERIALPRINT, the commands have no effect. Without any visible error message, I am not sure why this communication is failing. I have tested communication with the device (an arduino in this case) with another interface and am able to communicate with it.

**Comment by cfangmeier** [comment]
I don't think there's enough information here to diagnose the issue. The gScript serial commands are simple wrappers around the native LabVIEW serial interface and if those throw errors, they would get propagated up to the gScript layer. I checked how the wrapper is configured, and it is in "Synchronous mode", meaning that all the data should get pushed to the device immediately. So, It's not like some data is only getting partially transmitted and the rest sits in a buffer. It might help if you could share the gScript code that produces this as well as the Arduino sketch.

If we're not able to resolve this issue in a timely fashion, a band-aide could be to have the Arduino send some kind of confirmation message when it receives a command and then gScript checks for the confirmation and, if it's not there, re-send the command to the Arduino up to N times.

**Comment by shoogend** [comment]
I did a little more testing yesterday and I think the issue was on my end. Since the ports don't always stay the same (and I saw link in the gScript documentation for potentially alleviating this issue), I was iterating through the different available ports and connected to COM3. However, I think this was simply the wrong port for the arduino USB and so when I sent a signal through it, it of course had no effect. I disconnected and tried the remaining ports and connected to the one labelled ASRL5::INSTR, which was the correct one and I was able to communicate with the arduino properly. I will try setting up the enumeration so it is always the same to avoid connecting to the incorrect USB device.
